Sources/_MatchingEngine/Regex/AST/AST.swift

```diff
@@ -185,6 +188,43 @@ extension AST {
     }
   }
 
+  public struct AbsentFunction: Hashable, _ASTNode {
```

Comment:
I know Oniguruma uses this "absent function" name, but is there a better or more descriptive name? What does this do?

Reply:
Hmm, its primary role it to match against an inverse of a pattern (though the `.expression` variant also specifies what it can match). So maybe something like `MatchInverse` or `Exclusion`?

Reply:
Ah, I see. This is Oniguruma's approach to the regex-inversion problem (i.e. inverting each component of a concatenation is surprising and not the same as inverting the regex). I'm fine keeping this name with a comment for now, as a term-of-art or something to be renamed later. 

@rxwei any thoughts on how to model pattern inversions in the DSL?